sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -150,9 +150,9 @@ void analyze(boolean includeRules, SensorContext context) {
     Path analysisOutput = protobufReportPathForMSBuild12(context);
 
     try {
-      Files.write(analysisInput, analysisSettings.getBytes(Charsets.UTF_8));
+      Files.write(analysisInput, analysisSettings.getBytes(StandardCharsets.UTF_8));
     } catch (IOException e) {
-      throw Throwables.propagate(e);
+      throw new RuntimeException(e);
```

Comment:
We usually use IllegalStateException everywhere except if the exception should have a functional purpose (i.e. special handling somewhere). RuntimeException should do the trick as well.